system/CodeIgniter.php

```diff
@@ -290,6 +295,8 @@ protected function initializeKint()
      */
     public function run(?RouteCollectionInterface $routes = null, bool $returnResponse = false)
     {
+        assert($this->context !== null, 'Context must be set before run() is called');
```

Comment:
I'm not sure `assert()` is available outside of a dev context. Confirm?

Reply:
This check is not needed in production. Because it is not a runtime error,
but a logic error.

And php.ini decided assert() is available or not. 
See https://www.php.net/manual/en/ini.core.php#ini.zend.assertions

Reply:
Wow just read a lot on this, I was completely unaware of this whole area of PHP.